packages/fluentui/react-northstar/src/components/Carousel/Carousel.tsx

```diff
@@ -128,6 +139,24 @@ export const carouselSlotClassNames: CarouselSlotClassNames = {
   navigation: `${carouselClassName}__navigation`,
 };
 
+function useDirection(activeIndex) {
+  const prevActiveIndex = React.useRef<number>();
+  const directionRef = React.useRef<string>();
+
+  React.useMemo(() => {
+    if (activeIndex > prevActiveIndex.current) {
+      directionRef.current = 'right';
+    }
+    if (activeIndex < prevActiveIndex.current) {
+      directionRef.current = 'left';
+    }
+
+    prevActiveIndex.current = activeIndex;
+  }, [activeIndex]);
+
+  return directionRef.current;
+}
```

Comment:
```suggestion
function useDirection(activeIndex) {
  const prevActiveIndex = usePrevious<number>(activeIndex);
  const direction = React.useMemo(() => {
    if (activeIndex > prevActiveIndex.current) {
      return 'right';
    }
    if (activeIndex < prevActiveIndex.current) {
      return 'left';
    }
  }, [activeIndex]);

  return direction
}
```

- We can probably still use `usePrevious` there
- Usage of `React.useMemo()` can be transformed as `directionRef` looks redundant

Reply:
That is actually changing the behavior because in the current implementation L:154 we actually only update it on `activeIndex` changes no?

Reply:
I expect that it will be the same, did you test it?

Reply:
Yep, tried, doesn't work the same. It breaks animations if clicking somewhere that rerenders.

Reply:
I checked locally and both hooks return the same values. If the proposed implementation doesn't work, we need to debug it and fix as current implementation is an anti-pattern.

Reply:
Why antipattern?

They probably return same value, the problem is when you click in something and rerenders